exchanges/gateio/gateio_websocket_request_spot.go

```diff
@@ -261,18 +207,12 @@ func (g *Gateio) SendWebsocketRequest(ctx context.Context, channel string, connS
 		return fmt.Errorf("%s: %s", wsErr.Errors.Label, wsErr.Errors.Message)
 	}
 
-	to := struct {
-		Result any `json:"result"`
-	}{
-		Result: result,
-	}
-
-	return json.Unmarshal(inbound.Data, &to)
+	return json.Unmarshal(inbound.Data, &funnelResult{Result: result})
 }
 
-// InspectPayloadForAck checks the payload for an ack, it returns true if the
+// Inspect checks the payload for an ack, it returns true if the
 // payload does not contain an ack. This will force the cancellation of further
 // waiting for responses.
-func InspectPayloadForAck(data []byte) bool {
+func (g *Gateio) Inspect(data []byte) bool {
```

Comment:
Whoa 🙁 
Kinda cute to abuse GateIO as the Inspector type, but no matter how you name this, you can't pollute the GateIO namespace with a func named anything like this.
I was expecting a private type which would do `Appropriate`
```
type wsRespAckInspector struct {}
func (_ *wsRespAckInspector) Appropriate(m []byte) bool {
...
}
```

Reply:
Looks good.